app/controllers/api/v1/sessions_controller.rb

```diff
@@ -41,20 +41,27 @@ def session_params
       end
 
       def sign_in(user)
+        user.generate_session_token!(extended_session: session_params[:extend_session])
+
         # Creates an extended_session cookie if extend_session is selected in sign in form.
         if session_params[:extend_session]
           cookies.encrypted[:_extended_session] = {
             value: {
-              user_id: user.id
+              session_token: user.session_token,
+              session_expiry: user.session_expiry
```

Comment:
It seems that we're only writing the `:session_expiry` to  but not reading it from the cookie(s),
I think with this design we'd only need `session_digest` only?